### DISCUSSION TITLE
How is Thundernetes scaled?

### DISCUSSION THREAD

**Discussion by tomkerkhove** [PlayFab/thundernetes#173] [Q&A] [answered]
How is Thundernetes scaled? Do you use an app autoscaler? Is it KEDA?

**Comment by dgkanatsios** [comment, marked as answer]
thanks for reaching out! The GameServerBuild controller does the GameServer autoscaling. When you create a GameServerBuild, you specify how many standingBy and maximum servers you want. Thundernetes will do everything in its power to have this number available for you. As your matchmaker will allocate sessions, some of these standingBy servers will be converted to Active. Thundernetes will continue to create more standingBy servers till the sum of standingBy and Active will be equal to the maximum for this GameServerBuild.

Regarding cluster autoscaling, we recommend the official Kuberneters cluster autoscaler. Thundernetes helps by adding the "safe-to-evict" annotations to Pods that are standingBy, so they can be taken down during a cluster scale down.

Let us know if you need additional clarifications!

**Reply by tomkerkhove** [reply]
Interesting - So you have somewhat built your own autoscaler? Just so you know, [KEDA](http://keda.sh/) can be viable option to defer majority of that work and just use an external scaler so you can tell it when to scale.

Are you interested in that?

**Reply by dgkanatsios** [reply]
thanks for the recommendation @tomkerkhove. The autoscaling logic on Thundernetes is based exclusively on internal state (the GameServer state) and it's pretty simple in principle, the system tries to make sure there are enough servers in StandingBy state to cover the demand. 
Source: https://github.com/PlayFab/thundernetes/blob/main/pkg/operator/controllers/gameserverbuild_controller.go#L172
